src/test/java/com/google/finapp/FinAppTests.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.finapp;
+
+import com.google.cloud.ByteArray;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Key;
+import com.google.cloud.spanner.KeySet;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.UUID;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class FinAppTests {
+
+  private static SpannerDaoInterface JDBCDao;
+  private static SpannerDaoInterface JavaDao;
+  private static DatabaseClient databaseClient;
```

Comment:
This database client is passed into the Java client `SpannerDaoImpl` to use for its own mutations and used by the test suite to check the database for correct modifications. I think this should be fine, but please let me know if not.

Reply:
im not sure if this helps, but maybe [this](https://kkjavatutorials.com/how-to-mock-interface-using-mockito-example/) is worth checking out